bindings/pydeck-carto/pydeck_carto/carto_auth/auth.py

```diff
@@ -0,0 +1,217 @@
+import datetime
+import json
+import os.path
+
+import requests as requests
+
+
+class CredentialsError(Exception):
+    pass
+
+
+class CartoAuth:
+    def __init__(
+        self,
+        filepath=None,
+        client_id=None,
+        client_secret=None,
+        api_base_url="https://gcp-europe-west1.api.carto.com",
+        audience="carto-cloud-native-api",
+        grant_type="client_credentials",
+        cache_filepath=".carto_token.json",
+        access_token=None,
+        expire_in=None,
+        scope=None,
```

Comment:
I think that the `cache_filepath` should be kept as it could be configured by the user, maybe to have the cache in a different path, actually one of the tests is using different cache_filepaths to enforce this idea

Reply:
OK, let's keep it then